Create file price (xml,json)

prohoroff opened the pull request:
Test pull request)

prohoroff committed:
Create file price (xml,json)

zhurlik commented on price-calc/src/main/java/by/prohorov/price/kg/CostByPriceKgUseProperties.java:
```diff
@@ -33,8 +33,8 @@ public void setCostWeight(double costWeight) {
-            ClassLoader classLoader = CostByPriceKgUseFile.class.getClassLoader();
-            InputStream is = classLoader.getResourceAsStream("price_kg.properties");
+            ClassLoader classLoader = CostByPriceKgUseProperties.class.getClassLoader();
+            InputStream is = classLoader.getResourceAsStream("price/price_kg.properties");
```
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

prohoroff replied:
Hi. I'm corrected. Now I often use it.

zhurlik commented on price-calc/src/main/java/by/prohorov/price/entity/Price.java:
```diff
@@ -0,0 +1,41 @@
+
+public class Price {
+
+    private Integer light;
```
what happens when `Integer` will be nulls

prohoroff replied:
Created validatePrice

zhurlik commented on price-calc/src/main/java/by/prohorov/price/kg/CostByPriceKgUseFileXml.java:
```diff
@@ -0,0 +1,47 @@
+    public double costByPriceWeightInKg() {
+        try {
+            if (costWeight < 4) {
+                costWeight = Jackson.loaderXmlFile().getLight();
```
how will you handle the case when `Jackson` will not be able to load the needed data

prohoroff replied:
Add check Exception to class Jackson

zhurlik replied:
I don't like a lot of try-catch/exception, why you don't want for example set integers = 0 (by default), then you can return an empty `new Price()`. I mean you will not check nulls and will avoid unnecessary try-catch

prohoroff committed:
Create ValidatePrice
Create SelectPrice class for random use Price
Use only BigDecimal
Some update

prohoroff committed:
Refactor class Jackson
Create exception and use in code
Fix try-with-resources
Update validator price
Correct small mistakes

prohoroff committed:
Created more test
Updated exception
